airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceUtilities.java

```diff
@@ -71,6 +79,47 @@
     @Autowired
     private MediaFileService mediaFileService;
 
+    private static final Logger LOG = LoggerFactory.getLogger(SearchServiceUtilities.class);
+
+    private Random random;
+
+    {
+
+        String algorithmNative = "NativePRNG";
+        String algorithmSHA = "SHA1PRNG";
+
+        List<String> algorithms = Arrays.asList(Security.getProviders()).stream()
+                .flatMap(p -> p.getServices().stream().filter(s -> "SecureRandom".equals(s.getType())))
+                .map(s -> s.getAlgorithm()).collect(Collectors.toList());
+
+        if (algorithms.contains(algorithmNative)) {
+            /* linux */
+            try {
+                random = SecureRandom.getInstance(algorithmNative);
+                LOG.info("NativePRNG is used to create a random list of songs.", algorithmNative);
+            } catch (NoSuchAlgorithmException e) {
+                LOG.error("Usually unreachable.", e);
+            }
+        } else if (algorithms.contains(algorithmSHA)) {
+            /* win */
+            try {
+                random = SecureRandom.getInstance(algorithmSHA);
+                LOG.info("SHA1PRNG is used to create a random list of songs.");
+            } catch (NoSuchAlgorithmException e) {
+                LOG.error("Usually unreachable.", e);
+            }
+        }
+
+        /* others */
+        if (isEmpty(random)) {
```

Comment:
I don't think that testing for the list of algorithms being empty is safe here.
The list of algorithms could contain items that are not one of the two methods handled above, which would result in `random` being left un-initialized.
This should probably just be the fallthrough `else` on the above `if`.

Reply:
Exception cases are included.
It is different from ``else``.

Reply:
I know it is different.  My point is that it isn't safe to test for the list being empty as a fallthrough, as there can be other items that do **not** match the tests above.

Reply:
It is not a specification that you absolutely want to use something other than legacy.

Only one of the two popular algorithms are explicitly used.
Most environments other than the retro JVM are covered one or more of two algorithms.
For encryption requirements.

Even if advanced hardware has a special random number generator, it doesn't use it.
Taking the same method as legacy in an unexpectedly irregular environment is nothing dangerous or anything.

Reply:
It is not a specification that you absolutely want to use something other than legacy.

Only one of the two popular algorithms are explicitly used.
Most environments other than the retro JVM are covered one or more of two algorithms.
For encryption requirements.